bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/PS4Handler.java

```diff
@@ -0,0 +1,852 @@
+        sendPacketEncrypted(outPacket);
+    }
+
+    private void sendRemoteKey(int pushedKey) {
+        try {
+            SocketChannelHandler scHandler = socketChannelHandler;
+            int preWait = (scHandler == null || !loggedIn) ? POST_CONNECT_SENDKEY_DELAY_MS : 0;
+            SocketChannel channel = getConnection();
+
+            scheduler.schedule(() -> {
+                ByteBuffer outPacket = PS4PacketHandler.makeRemoteControlPacket(PS4_KEY_OPEN_RC);
+                sendPacketEncrypted(outPacket, channel);
+
+                scheduler.schedule(() -> {
+                    // Send remote key
+                    ByteBuffer keyPacket = PS4PacketHandler.makeRemoteControlPacket(pushedKey);
+                    sendPacketEncrypted(keyPacket, channel);
+
+                    scheduler.schedule(() -> {
+                        ByteBuffer offPacket = PS4PacketHandler.makeRemoteControlPacket(PS4_KEY_OFF);
+                        sendPacketEncrypted(offPacket, channel);
+
+                        scheduler.schedule(() -> {
+                            ByteBuffer closePacket = PS4PacketHandler.makeRemoteControlPacket(PS4_KEY_CLOSE_RC);
+                            sendPacketEncrypted(closePacket, channel);
+                        }, MIN_SENDKEY_DELAY_MS, TimeUnit.MILLISECONDS);
+
+                    }, MIN_HOLDKEY_DELAY_MS, TimeUnit.MILLISECONDS);
+
+                }, MIN_SENDKEY_DELAY_MS, TimeUnit.MILLISECONDS);
+
+            }, preWait, TimeUnit.MILLISECONDS);
+
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void parseSearchResponse(DatagramPacket packet) {
+        byte[] data = packet.getData();
+        String message = new String(data, StandardCharsets.UTF_8);
+        String applicationName = "";
+        String applicationId = "";
+
+        String[] rowStrings = message.trim().split("\\r?\\n");
+        for (String row : rowStrings) {
+            int index = row.indexOf(':');
+            if (index == -1) {
+                OnOffType power = null;
+                if (row.contains("200")) {
+                    power = OnOffType.ON;
+                } else if (row.contains("620")) {
+                    power = OnOffType.OFF;
+                }
+                if (power != null) {
+                    updateState(CHANNEL_POWER, power);
+                    if (!currentPower.equals(power)) {
+                        currentPower = power;
+                        if (power.equals(OnOffType.ON) && config.autoConnect) {
+                            SocketChannelHandler scHandler = socketChannelHandler;
+                            if (scHandler == null || !loggedIn) {
+                                logger.debug("Trying to login after power on.");
+                                scheduler.schedule(() -> login(), 20, TimeUnit.SECONDS);
```

Comment:
At this point I think you might as well manage your own scheduler so that you can destroy any pending tasks a part of handler disposal. It would certainly be easier than trying to keep track of all of these scheduled task futures.